src/app/static/src/app/store/adrUpload/actions.ts

```diff
@@ -91,9 +91,10 @@ export const actions: ActionTree<ADRUploadState, RootState> & ADRUploadActions =
 
     async uploadFilesToADR(context, uploadFilesPayload) {
         const {rootState, commit, dispatch} = context;
-        const uploadMetadata = rootState.modelRun.result?.uploadMetadata
         const selectedDatasetId = rootState.baseline.selectedDataset!.id;
-        const modelCalibrateId = rootState.modelCalibrate.calibrateId;
+        let downloadId;
+
+        const uploadMetadata = rootState.metadata.adrUploadMetadata
```

Comment:
Is there any chance of a race condition here? Should this function be invoking and waiting for `getUploadMetadata` rather than `DownloadResults` doing it pre-emptively? The latter seems to be mixing download and upload concerns.